src/main/java/uk/gov/justice/hmpps/casenotes/notes/internal/ReadCaseNote.kt

```diff
@@ -0,0 +1,55 @@
+package uk.gov.justice.hmpps.casenotes.notes.internal
+
+import org.springframework.data.domain.Page
+import org.springframework.data.domain.PageRequest
+import org.springframework.data.domain.Pageable
+import org.springframework.data.domain.Sort
+import org.springframework.data.domain.Sort.by
+import org.springframework.security.access.prepost.PreAuthorize
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+import uk.gov.justice.hmpps.casenotes.config.SecurityUserContext.Companion.ROLE_CASE_NOTES_READ
+import uk.gov.justice.hmpps.casenotes.config.SecurityUserContext.Companion.ROLE_CASE_NOTES_WRITE
+import uk.gov.justice.hmpps.casenotes.dto.CaseNoteFilter
+import uk.gov.justice.hmpps.casenotes.notes.CaseNote
+import uk.gov.justice.hmpps.casenotes.notes.asLegacyId
+import uk.gov.justice.hmpps.casenotes.services.EntityNotFoundException
+import java.util.UUID.fromString
+
+@Service
+@Transactional(readOnly = true)
+@PreAuthorize("hasAnyRole('$ROLE_CASE_NOTES_READ', '$ROLE_CASE_NOTES_WRITE')")
+class ReadCaseNote(
+  private val noteRepository: NoteRepository,
+) {
+  fun caseNotes(prisonNumber: String, filter: CaseNoteFilter, pageable: Pageable): Page<CaseNote> =
+    noteRepository.findAll(filter.asSpecification(prisonNumber), pageable.forSpecification()).map { it.toModel() }
+
+  fun caseNote(prisonNumber: String, caseNoteId: String): CaseNote {
+    val caseNote = when (val legacyId = caseNoteId.asLegacyId()) {
+      null -> noteRepository.findByIdAndPrisonNumber(fromString(caseNoteId), prisonNumber)
+      else -> noteRepository.findByLegacyIdAndPrisonNumber(legacyId, prisonNumber)
+    } ?: throw EntityNotFoundException.withId(caseNoteId)
+    return caseNote.toModel()
+  }
+
+  companion object {
+    const val SOURCE = "OCNS"
+  }
+}
+
+private fun CaseNoteFilter.asSpecification(prisonNumber: String) =
+  listOfNotNull(
+    matchesPrisonNumber(prisonNumber),
+    matchesOnType(includeSensitive),
+    locationId?.let { matchesLocationId(it) },
+    authorUsername?.let { matchesAuthorUsername(it) },
+    startDate?.let { occurredAfter(it) },
+    endDate?.let { occurredBefore(it) },
+  ).reduce { spec, current -> spec.and(current) }
+
+private fun Pageable.forSpecification(): Pageable {
```

Comment:
Guessing this is to maintain compatibility with the controller's pageable property while still restricting down to the only (currently) valid ordering options